src/controllers/cartController.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable no-unreachable */
+/* eslint-disable radix */
+import {Product, cart} from "../database/models";
+
+class CartController {
+  static async addItem(req, res) {
+    try {
+      const {productId} = req.params;
+      const product = await Product.findOne({ where: { productId, available: true } });
+       if (!product) {
+        return res.status(404).json({ status:"not Found", message: "Product not found" });
+        }
+        const item = {
+            productId: product.productId,
+            productName: product.productName,
+            productImage: product.productImage[0],
+            quantity: 1,
+            productPrice: product.productPrice,
+            productTotal: product.productPrice,
+          };
+        const Cart = await cart.findOne({ where:{ buyerId:req.user.id } });
+        if (!Cart) {
+
+        const newCart = await cart.create({
+          products:[item],
+          buyerId: req.user.id,
+        });
+        newCart.total = newCart.products .map((item1) => JSON.parse(item1.productTotal)).reduce((sum, next) => sum + next);
+        await newCart.save();
+        return res.status(201).json({ status: "success", message: "product added to cart successfully" });
+    }   
+        const itemExists = Cart.products.some((cartItem) => cartItem.productId === product.productId);
+        if (itemExists === true) {
+        return res.status(400).json({ status: "fail", message: "Product already in cart" });
+      }
+        Cart.products.push(item);
+        const subtotal = Cart.products.map((item1) => JSON.parse(item1.productTotal)).reduce((sum, next) => sum + next);
+        await cart.update(
+          { products: Cart.products, total: subtotal },
+          { where: { id: Cart.id } }
+        );
+        res.status(201).json({ status: "success", message: "product added to cart successfully" });
+      }
+     
+        catch (error) {
+        console.error(error);
+        if (error.name === "SequelizeValidationError") {
+        return res.status(422).json({ status: "DatabaseError", message: error.message });
+      }
+        res.status(500).json({message: "Internal server error" });
+    }
+  };
+
+  static async getCartItems(req, res) {
+    try {
+      const Cart = await cart.findOne({ where: { buyerId: req.user.id } });
+      if (!Cart) {
+        res.status(401).json({ status: "fail", message: "Cart is Empty" });
+      } else {
+        res.status(304).json({ status:"success", Cart });
```

Comment:
- return correct response

```suggestion
        res.status(200).json({ status:"success", message: "cart items", data:Cart });
```